src/org/labkey/test/tests/SampleTypeTest.java

```diff
@@ -170,6 +170,51 @@ public void testCreateSampleTypeNoExpression()
         sampleTypeHelper.verifyDataValues(data);
     }
 
+    // Issue 47280: LKSM: Trailing whitespace in Source name won't resolve when deriving samples
+    @Test
+    public void testImportSamplesWithTrailingSpace()
+    {
+        final String sampleTypeName = "SampleTypeWithProvidedName";
+        final List<FieldDefinition> fields = List.of(
+                new FieldDefinition("IntCol", FieldDefinition.ColumnType.Integer),
+                new FieldDefinition("StringCol", FieldDefinition.ColumnType.String),
+                new FieldDefinition("DateCol", FieldDefinition.ColumnType.DateAndTime),
+                new FieldDefinition("BoolCol", FieldDefinition.ColumnType.Boolean));
+
+        SampleTypeDefinition sampleTypeDefinition = new SampleTypeDefinition(sampleTypeName).setFields(fields);
+
+        log("Create a new sample type with no name expression");
+        projectMenu().navigateToFolder(PROJECT_NAME, FOLDER_NAME);
+        SampleTypeHelper sampleTypeHelper = new SampleTypeHelper(this);
+        sampleTypeHelper.createSampleType(sampleTypeDefinition);
+        sampleTypeHelper.goToSampleType(sampleTypeName);
+
+        log("Add a single row to the sample type, with trailing spaces");
+        Map<String, String> fieldMap = Map.of("Name", "S-1 ", "StringCol", "Ess ", "IntCol", "1 ");
+        sampleTypeHelper.insertRow(fieldMap);
```

Comment:
Should we also test for leading spaces while we're at it?  

How about a test for renaming that adds a space?